File: server/kafka-backend/src/core/domain/events/transfer-initiated.event.ts
export class TransferInitiatedEvent
{
  constructor(
    public readonly trackingId: string,
    public readonly fromAccountId: string,
    public readonly toAccountId: string,
    public readonly fromIban: string,
    public readonly toIban: string,
    public readonly amount: number,
    public readonly currency: string,
    public readonly description: string,
    public readonly timestamp: Date = new Date(),
  ) {}

  static eventName = 'transfer.initiated';
}

File: server/java-backend/src/main/java/com/intbank/core/domain/event/TransferInitiatedEvent.java
package com.intbank.core.domain.event;

import java.time.Instant;

public record TransferInitiatedEvent(
    String trackingId,
    String fromAccountId,
    String toAccountId,
    String fromIban,
    String toIban,
    double amount,
    String currency,
    String description,
    Instant timestamp
)
{
    public static final String EVENT_NAME = "transfer.initiated";
}